src/main.rs

```diff
@@ -209,24 +218,28 @@ fn set_new_password(db_conn: &str, uname: &str, password_hash: &str) -> Result<(
     }
 }
 
-fn hash_password(password: &str, pepper: &str) -> String {
+fn hash_password(password: &str, pepper: &str, rounds: u32) -> String {
     // This function closely mimics
     // https://github.com/matrix-org/synapse/blob/9bba6ebaa903a81cd94fada114aa71e20b685adb/scripts/hash_password
-
-    // Match the default from synapse arbitrarily
-    let bcrypt_rounds = 12;
-
     let mut salt = [0u8; 16];
     let mut rng = OsRng::new().unwrap();
     rng.fill_bytes(&mut salt[..]);
     let mut output = [0u8; 24];
 
-    bcrypt::bcrypt(bcrypt_rounds,
+    // ... See https://github.com/pyca/bcrypt/blob/fcebaa0db74dc822877128e57a79dcfda2a2dc4f/src/bcrypt/__init__.py#L66-L72
+    // and https://github.com/DaGenix/rust-crypto/blob/cc1a5fde1ce957bd1a8a2e30169443cdb4780111/src/bcrypt.rs#L26
+    let peppered_password = format!("{}{}", password, pepper);
+    let bcryptable_password = &peppered_password.as_bytes()[0..72];
```

Comment:
It's bad yes.

But it's more important to match the python implementation since, well, the whole point of this is to write to a DB which will be read by python and checked by python.

So fixing this means it won't work